analyzers/src/SonarAnalyzer.CSharp/Extensions/SyntaxNodeExtensions.cs

```diff
@@ -31,18 +31,21 @@ internal static partial class SyntaxNodeExtensions
             SyntaxKind.AddAccessorDeclaration,
             SyntaxKind.AnonymousMethodExpression,
             SyntaxKind.BaseConstructorInitializer,
+            SyntaxKind.CompilationUnit,
             SyntaxKind.ConstructorDeclaration,
             SyntaxKind.ConversionOperatorDeclaration,
             SyntaxKind.DestructorDeclaration,
-            SyntaxKind.EqualsValueClause,
+            SyntaxKind.EnumMemberDeclaration,
+            SyntaxKind.FieldDeclaration,
             SyntaxKind.GetAccessorDeclaration,
-            SyntaxKind.GlobalStatement,
```

Comment:
GlobalStatement is not really a scope. CompilationUnit is the better fit. (EnclosingScopeSyntaxKinds needs to be as close to what [RegisterCodeBlockAction](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.diagnostics.analysiscontext.registercodeblockaction?view=roslyn-dotnet-4.7.0) does.)